usaspending_api/common/helpers/agency_logic_helpers.py

```diff
@@ -0,0 +1,39 @@
+from usaspending_api.download.lookups import CFO_CGACS
+from usaspending_api.references.constants import (
+    DOD_SUBSUMED_CGAC,
+    DOD_CGAC,
+    DOD_ARMED_FORCES_TAS_CGAC_FREC,
+    DHS_CGAC,
+    DHS_SUBSUMED_CGAC,
+)
+from usaspending_api.references.models import ToptierAgency
+
+
+def agency_from_identifiers(cgac, frec):
+    """
+    General logic for what agency to include a given CGAC/FREC combination under, accounting for all rollups.
+    :param cgac:
+    :param frec:
+    :return: string, which is the toptier_code for the correct agency to count this combination under
+    """
+    if (cgac, frec) in DOD_ARMED_FORCES_TAS_CGAC_FREC:
+        return DOD_CGAC
+
+    if cgac in DOD_SUBSUMED_CGAC:
+        return DOD_CGAC
+
+    if cgac in DHS_SUBSUMED_CGAC:
+        return DHS_CGAC
```

Comment:
Is this how we're handling FEMA as a toptier now?  Previously, it was treated as a toptier but given lower visibility.  Is the goal to not show FEMA as a toptier in the top tier of the TAS control or is this a new site-wide thing?

Reply:
Ross said, "any 058 federal account would be listed under 070 for the TAS filter" in conversations for clarification. Combined with the requirement to not show agencies without TAS under them, I took the interpretation that FEMA should not be shown at all.

Reply:
I wish we had a place to record such decisions.  I've implemented many off-the-cuff decisions like this.  Wonder if this needs to apply anywhere else or just in the tree.  Thanks for the clarification.

Reply:
These types of decisions heard from the PO, whether coming to _any_ of the tester, developer, or scrum master, **should be recorded as _acceptance criteria_** in the ticket, and re-affirmed by the PO and acknowledged by the tester by way of Approving the AC (-> _Ready for Sprint_)

Reply:
I've more than once had a chat with the PO that ended with, "Talk to Kirk, he knows this stuff". We're moving into retro territory, but the reality is that this business logic is fluid. The code above, in fact, I know to be wrong now.

Reply:
We're in deep trouble if I'm the source of business rules.  I wish I knew a good way to record these things.  I've been on a couple of projects where we've attempted to capture business rules, but they've never really worked very well (in my opinion).  The closest we came was a Lotus Notes repository where we organized business rules.  Is Lotus Notes still a thing?  Man I'm old.